src/sardana/spock/spockms.py

```diff
@@ -602,3 +602,32 @@ def _removeMacro(self, macro_info):
         macro_name = macro_info.name
         genutils.unexpose_magic(macro_name)
         del self._local_magic[macro_name]
+
+class Reflector(object):
+    def __getitem__(self, name):
+        return name
+
+def split_macro_parameters(parameters_s):
+    macro_params = genutils.arg_split(parameters_s, posix=True)
+
+    list_of_pars = 0
+    if len(macro_params) == 1:
+        if  macro_params[0].find('[') > -1:
+            list_of_pars = 1
+    # skip empty strings and just one parameter
+    if len(macro_params) > 1 or list_of_pars:
+        params_str = ''
+        for par in macro_params:
+            params_str = params_str + str(par) + ","
+        params_str = params_str[:-1]
+        params_str = params_str.replace("[,", "[").replace(",]", "]").replace(
+            "][", "],[")
+        macro_params = eval(params_str, globals(), Reflector())
```

Comment:
Using direct eval may be dangerous. Please use [SafeEvaluator](https://github.com/taurus-org/taurus/blob/develop/lib/taurus/core/util/safeeval.py#L35) instead which takes care of allowing only safe code to be evaluated.